compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/symbols/BSymbol.java

```diff
@@ -46,6 +46,7 @@
     public boolean tainted;
     public boolean closure;
     public MarkdownDocAttachment markdownDocumentation;
+    public boolean isDeprecated;
```

Comment:
This implementation will work only with the same module. Because we don't persist this with BALO. Easy wat to implement this makes Deprecated as a part of the symbol flag.

Reply:
Done